conversion/conversion_test.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversion_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cloudspannerecosystem/harbourbridge/conversion"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/stretchr/testify/assert"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var (
```

Comment:
The code in this file is very similar to the integrations_test.go files under testing.

I'm wondering if organizationally it might be better to put this test under that tree e.g. in a directory something like testing/conversion. The reason is that all of the similar tests are there. Note that the tests we tend to put with regular code (e.g. internal/convert_test.go) are typically simple unit tests that are self contained and don't rely on having a cloud spanner instance (or emulator) around.

Also, we should probably put a TODO somewhere in this file to do a refactoring and remove all of the redundant code we're accumulating in the other integration_test.go files and in this file -- we should have some kind of test utils with most of the core code in.